doc/manual/manual.tex

```diff
@@ -4732,6 +4732,100 @@ \subsubsection{Averaging material properties}
 \end{table}
 
 
+\subsubsection{Prescribed internal velocity constraints}
+\label{sec:prescribed-velocities}
+\textit{This section was contributed by Jonathan Perry-Houts}
+
+In cases where it is desirable to investigate the behavior of one part of the model
+domain but the controlling physics of another part is difficult to capture,
+such as corner flow in subduction zones, it may be useful to force the desired
+behavior in some parts of the model domain and solve for the resulting flow
+everywhere else. This is possible through the use of \aspect{} ``signals'' mechanism.
+
+Internally, \aspect{} adds ``constraints'' to the finite element system for boundary conditions and hanging nodes. These are places in the finite element system where certain solution variables are required to match some prescribed value. Although it is somewhat mathematically inadmissible to prescribe constraints on nodes inside the model domain, $\Omega$, it is nevertheless possible to do so so long as our prescribed velocity field fits in to the finite element's solution space.
+
+Through \aspect{}'s signals mechanism (as documented in Section~\ref{sec:extending-signals}). We need to write a shared library which provides a ``socket'' for the signal which is triggered after the regular model constraints are assigned, but before they are ``distributed.''
+
+A more complete implementation of this functionality is available in \url{cookbooks/prescribed-velocity/}, but a working skeleton for similar functionality is provided below.
```

Comment:
ah, than the one below?

I wouldn't mind if we just showed the whole .cc file here (via `\lstinputlisting`)